services/node_manager/src/main/java/com/futurewei/alcor/nodemanager/dao/NodeRepository.java

```diff
@@ -103,6 +105,19 @@ public void addItem(NodeInfo nodeInfo) throws CacheException {
         }
     }
 
+    @Override
+    public void addItems(List<NodeInfo> items) throws CacheException {
+        try(Transaction tx =cache.getTransaction().start()){
+            Map<String, NodeInfo> nodeInfoMap = items.stream().collect(Collectors.toMap(NodeInfo::getId, Function.identity()));
+            cache.putAll(nodeInfoMap);
+            tx.commit();
+        } catch (CacheException e) {
```

Comment:
@songxiaoyan Here we may hide an exception if it is not CacheException. 

Try to apply the same exception handling in services/route_manager/src/main/java/com/futurewei/alcor/route/dao/RouteEntryRepository.java

Reply:
I used to refer to the exception thrown by the other method in each class, But now I have modified these as discussed with Piaoping. No transaction required, all exceptions are thrown.